### PR TITLE
AK+Everywhere: Replace DistinctNumeric bool parameters with named ones

### PR DESCRIPTION
**Pull request by AtkinsSJ**
This means that rather than this:

```c++
AK_TYPEDEF_DISTINCT_NUMERIC_GENERAL(u64, true, true, false, false, false, true, FunctionAddress);
```

We now have this:
```c++
AK_TYPEDEF_DISTINCT_NUMERIC_GENERAL(u64, FunctionAddress, Arithmetic, Comparison, Increment);
```

Which is a lot more readable. :^)

cc: @alimpfard

### CODE REVIEW

File: AK/DistinctNumeric.h [hunk 1]
```diff
@@ -266,29 +303,33 @@ class DistinctNumeric {
...
-    using NAME = DistinctNumeric<T, struct __##NAME##_tag, Incr, Cmp, Bool, Flags, Shift, Arith>;
-#define AK_TYPEDEF_DISTINCT_ORDERED_ID(T, NAME) AK_TYPEDEF_DISTINCT_NUMERIC_GENERAL(T, false, true, true, false, false, false, NAME)
+#define AK_TYPEDEF_DISTINCT_NUMERIC_GENERAL(T, NAME, ...)                                       \
+    struct NAME##_decl {                                                                        \
+        using Arithmetic [[maybe_unused]] = AK::DistinctNumericFeature::Arithmetic;             \
+        using BoolCast [[maybe_unused]] = AK::DistinctNumericFeature::BoolCast;                 \
+        using Comparison [[maybe_unused]] = AK::DistinctNumericFeature::Comparison;             \
+        using Flags [[maybe_unused]] = AK::DistinctNumericFeature::Flags;                       \
+        using Increment [[maybe_unused]] = AK::DistinctNumericFeature::Increment;               \
+        using Shift [[maybe_unused]] = AK::DistinctNumericFeature::Shift;                       \
```
Comment (alimpfard) [inline AK/DistinctNumeric.h:322]: You can replace all of this with `using namespace AK::DistinctNumericFeature;`

File: AK/DistinctNumeric.h [hunk 2]
```diff
@@ -266,29 +303,33 @@ class DistinctNumeric {
...
+        using NAME [[maybe_unused]] = DistinctNumeric<T, struct __##NAME##_tag, ##__VA_ARGS__>; \
+    };                                                                                          \
+    using NAME = NAME##_decl::NAME;
+#define AK_TYPEDEF_DISTINCT_ORDERED_ID(T, NAME) AK_TYPEDEF_DISTINCT_NUMERIC_GENERAL(T, NAME, Comparison, BoolCast)
```
Comment (alimpfard) [inline AK/DistinctNumeric.h:326]: Leave an empty line here, might just be me, but took a second to parse

**Review by alimpfard — COMMENTED** [review]
Very nice!
couple notes:

**Comment by AtkinsSJ** [comment]
Dang it, the error I was seeing in CLion showed up in CI.
https://dev.azure.com/SerenityOS/SerenityOS/_build/results?buildId=22079&view=logs&j=1792fc0c-aa2f-5547-0702-e8eb3cee2177&t=419c1473-5af3-5c24-9e2e-2e8533c287ae&l=5005

File: AK/DistinctNumeric.h (hunk 1 above)
Comment (alimpfard) [reply AK/DistinctNumeric.h]: oh right, you made that a struct...nevermind then.

**Comment by alimpfard** [comment]
>this wants a typename inserted in there for some reason, but not always?

Dependent names have been somewhat relaxed starting with C++20, but explicit `typename`/`template`s are sometimes still needed - I suggest always keeping them there.

**Comment by AtkinsSJ** [comment]
I've added the `typename` keyword where it wanted it. I'm sure I tried that before and it gave me different errors, but this time it seems happy. :yakshrug:

**Pull request merged**